# Problem:
Create a model to predict the possibility if the match on a day will be played or not based on the details of Temperature, Humidity and Rain conditions. 

In [1]:
import pandas as pd
from sklearn.tree import DecisionTreeClassifier
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import classification_report
from sklearn.model_selection import train_test_split

In [2]:
dt=pd.read_csv('data/play_stats.csv')
dt.head()

Temperature  Humidity% Rain Play
0           34       74.2  Yes   No
1           19       68.2   No   No
2           28       67.2  Yes  Yes
3           29       66.6  Yes   No
4           26       57.9  Yes  Yes

In [3]:
Enc=LabelEncoder()
Enc.fit(['Yes','No'])
for col in ['Rain','Play']:
    dt[col]=Enc.transform(dt[col])

In [4]:
X=dt.drop(columns='Play')
y=dt['Play']
X_train,X_test,Y_train,Y_test=train_test_split(X,y,test_size=0.1)
Cmodel=DecisionTreeClassifier()
Cmodel.fit(X_train,Y_train)

DecisionTreeClassifier()

In [5]:
predY=Cmodel.predict(X_test)
print(classification_report(Y_test,predY))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        80
           1       1.00      1.00      1.00        21

    accuracy                           1.00       101
   macro avg       1.00      1.00      1.00       101
weighted avg       1.00      1.00      1.00       101



In [6]:
def predict_play():
    temp=int(input("Temperature:"))
    humid=int(input("Humidity:"))
    rain=input("Rain:(Yes/No)")
    rain=Enc.transform([rain])
    #Ask the model for prediction
    print(temp,humid,rain[0])
    pred=Cmodel.predict([[temp,humid,rain[0]]])
    # Decoding the prediction to Label
    rec=Enc.inverse_transform(pred)
    return rec[0] 

In [7]:
predict_play()

19 68 0


/Users/jigishap/.pyenv/versions/Movie_Recommender/lib/python3.10/site-packages/sklearn/base.py:420: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names
  warnings.warn(


'No'

With temperature= 19 units, Humidity= 68%, Rain=No , the model predicts that the match will be played